miden-tx/src/testing/tx_context.rs

```diff
@@ -204,14 +251,41 @@ impl TransactionContextBuilder {
         self
     }
 
-    /// Populates input and expected notes with the results from [mock_notes()]
+    /// Populates input and expected notes.
     pub fn with_mock_notes(self, asset_preservation: AssetPreservationStatus) -> Self {
-        let (notes, output_notes) = mock_notes(&self.assembler, &asset_preservation);
+        let (mut input_notes, output_notes) = mock_notes(&self.assembler);
+
+        let consumed_note_5 = input_notes.pop().unwrap();
+        let consumed_note_4 = input_notes.pop().unwrap();
+        let consumed_note_3 = input_notes.pop().unwrap();
+        let consumed_note_2 = input_notes.pop().unwrap();
+        let consumed_note_1 = input_notes.pop().unwrap();
+
+        let notes = match asset_preservation {
+            AssetPreservationStatus::TooFewInput => vec![consumed_note_1],
+            AssetPreservationStatus::Preserved => {
+                vec![consumed_note_1, consumed_note_2]
+            },
+            AssetPreservationStatus::PreservedWithAccountVaultDelta => {
+                vec![consumed_note_1, consumed_note_2, consumed_note_5]
+            },
+            AssetPreservationStatus::TooManyFungibleInput => {
+                vec![consumed_note_1, consumed_note_2, consumed_note_3]
+            },
+            AssetPreservationStatus::TooManyNonFungibleInput => {
+                vec![consumed_note_1, consumed_note_2, consumed_note_4]
+            },
+        };
+
         self.input_notes(notes).expected_notes(output_notes)
     }
```

Comment:
I need to write tests with different notes. My motivation for this cleanup was to make that possible. The code is still not there, the note creation needs to be further split. The goal here was at least to get the note creation close to the context, so that eventually these can be created via the mock_chain instead of `AssetPreservationStatus`.

I think the `AssetPreservationStatus` is in general a bad idea. As it is, every different scenario will need a flag in that enum, which means:

- `mock_notes` will create the superset of the notes, which is already making this function grow too much.
- The enum above will list all of the possible scenarios. Which is cumbersome, so we are not adding new options, and reusing what exist, and that means it is hard to figure what is the purpose of these scenarios because they are shared among dozens of tests.

Ideally the note creation should be pushed to the tests, via the use of the mock chain. But that will have to be in another PR. I don't know if it is best to place it in `TransactionContextBuilder` or `TransactionContext` or somewhere else.